enhancements/machine-api/machine-api-usage-telemetry.md

```diff
@@ -0,0 +1,338 @@
+automated remediation.
+
+#### Story 3
+
+As an OpenShift developer I would like to correlate Machine API resource usage
+with alerting events to provide presciptive advice about failures conditions
+and platform instability. By examining data about Machine API resource management,
+failures, and remediations, and corelating that data with alerts and platform
+information I will be able to better advise users about errors and methods for
+avoiding them. When combined with a wider fleet view this information will help
+me to create advisories about provider instabilities and potential outtages.
+
+### Implementation Details/Notes/Constraints
+#### Specific Telemetry Inquiries
+
+This section details some of the specific use cases that have been identified
+as initial inquries we would like to gather and report on. They have been
+categorized into 3 epics: demonstration of value, engineering insights,
+and proactive support. Demonstration of value speaks to the raw amount of
+usage that is happening for any given component. Engineering insights speaks
+to situations where we can learn about user experience and interactions to inform
+future planning and guidance. Proactive support speaks to information that we
+learn which will lead to better experiences for users by providing opportunity
+for preventitive intervention.
+
+**Total Remediation Events**
+
+Epics: Demonstration of Value, Proactive Support
+
+This metric will give us insight into how effectively the MHC is being used.
+A sustained high number of events on a particular cluster will enable us to
+deliver proactive support. A sudden spike across all clusters might give us
+early warning to a cloud provider’s data center outage, or a large network
+or similar event that is impacting many of our users.
+
+For example, since many clusters operate at a relatively small scale, it’s
+possible that losing an AZ in US-East-1 would be small enough event on their
+cluster to trigger MHC. Letting users know we’re seeing this across many
+clusters would assist in their own troubleshooting efforts and save them time.
+
+Having a total number of remediation events can demonstrate the value our
+platform is bringing to our users. It will show the operational burden we’ve
+decreased on teams that use OpenShift.
+
+**Remediation is Disabled or Short Circuited**
+
+Epics: Proactive Support
+
+This metric will give us insight into when a particular cluster has too many
+unhealthy nodes for the MHC to do its job. A user may be expecting the MHC is
+running as normal, cleaning up broken hosts, when in reality it has been silently disabled.
+Any sustained disabling of the MHC should potentially trigger a proactive
+support engagement. Similar to the _Total Remediation Events_, if any clusters
+experience this simultaneously, we can help provide any potential outage info
+to our users.
+
+**Number of Nodes Covered by the Machine Health Check**
```

Comment:
How would we know the size of the cluster? Is this kind of data already available?

Reply:
we can calculate the number of nodes from metrics currently exposed, for example [`cluster:node_instance_type_count:sum`](https://github.com/openshift/cluster-monitoring-operator/blob/master/manifests/0000_50_cluster_monitoring_operator_04-config.yaml#L113). there might be other ways to figure this as well, but with the `node_instance_type_count` we could gather all the nodes by disregarding the cardinality of the instance types and just summing all counts.